tests/integration/integration_python_test.go

```diff
@@ -1880,6 +1888,33 @@ func TestPythonComponentProviderGetSchema(t *testing.T) {
 	component := resources["provider:index:MyComponent"].(map[string]interface{})
 	require.NoError(t, json.Unmarshal([]byte(expectedJSON), &expected))
 	require.Equal(t, expected, component)
+
+	// Check the complex types
+	expectedTypesJSON := `{
+		"provider:index:Complex": {
+	      "properties": {
+	        "complexStr": {
+	          "type": "string"
+	        },
+	        "nested": {
+	          "$ref": "#/types/provider:index:Nested"
+	        }
+	      },
+	      "type": "object"
+	    },
+	    "provider:index:Nested": {
+	      "properties": {
```

Comment:
Is `"required"` missing? All types seem to have properties required.

Reply:
Yes, good catch! I missed this when building the schema. Fixed in https://github.com/pulumi/pulumi/pull/18419/commits/04f33d7f1ff755b466d68a191ec7c2f90404cb0b